src/model/user.rs

```diff
@@ -185,17 +185,21 @@ impl CurrentUser {
     /// use serenity::model::permissions::*;
     ///
     /// // assuming the cache has been unlocked
-    /// let url = cache.user.invite_url(READ_MESSAGES | SEND_MESSAGES | EMBED_LINKS);
+    /// let url = cache.user.invite_url(READ_MESSAGES | SEND_MESSAGES | EMBED_LINKS).unwrap();
     ///
     /// assert_eq!(url, "https://discordapp.com/api/oauth2/authorize?client_id=249608697955745802&scope=bot&permissions=19456");
     /// ```
-    pub fn invite_url(&self, permissions: Permissions) -> String {
+    pub fn invite_url(&self, permissions: Permissions) -> Result<String> {
```

Comment:
Can we make a note in the primary description that this hits REST, and list in an errors section the possible errors (`Unauthorized`; see `ChannelId::send_file` for an example on how to format it)?